src/plugins/dashboard/public/url_utils/url_helper.test.ts

```diff
@@ -24,16 +24,17 @@ describe('', () => {
     const id = '123eb456cd';
     const url =
       "/pep/app/dashboards#/create?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(description:'',filters:!())";
-    expect(addEmbeddableToDashboardUrl(url, id, 'visualization')).toEqual(
-      `/pep/app/dashboards#/create?_a=%28description%3A%27%27%2Cfilters%3A%21%28%29%29&_g=%28refreshInterval%3A%28pause%3A%21t%2Cvalue%3A0%29%2Ctime%3A%28from%3Anow-15m%2Cto%3Anow%29%29&addEmbeddableId=${id}&addEmbeddableType=visualization`
+
+    expect(addEmbeddableToDashboardUrl(url, id, 'visualization')).toBe(
+      '/pep/app/dashboards?addEmbeddableId=123eb456cd&addEmbeddableType=visualization#%2Fcreate%3F_g%3D%28refreshInterval%3A%28pause%3A%21t%2Cvalue%3A0%29%2Ctime%3A%28from%3Anow-15m%2Cto%3Anow%29%29%26_a%3D%28description%3A%27%27%2Cfilters%3A%21%28%29%29'
     );
   });
   it('addEmbeddableToDashboardUrl when dashboard is saved', () => {
     const id = '123eb456cd';
     const url =
       "/pep/app/dashboards#/view/9b780cd0-3dd3-11e8-b2b9-5d5dc1715159?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(description:'',filters:!())";
-    expect(addEmbeddableToDashboardUrl(url, id, 'visualization')).toEqual(
-      `/pep/app/dashboards#/view/9b780cd0-3dd3-11e8-b2b9-5d5dc1715159?_a=%28description%3A%27%27%2Cfilters%3A%21%28%29%29&_g=%28refreshInterval%3A%28pause%3A%21t%2Cvalue%3A0%29%2Ctime%3A%28from%3Anow-15m%2Cto%3Anow%29%29&addEmbeddableId=${id}&addEmbeddableType=visualization`
+    expect(addEmbeddableToDashboardUrl(url, id, 'visualization')).toBe(
+      '/pep/app/dashboards?addEmbeddableId=123eb456cd&addEmbeddableType=visualization#%2Fview%2F9b780cd0-3dd3-11e8-b2b9-5d5dc1715159%3F_g%3D%28refreshInterval%3A%28pause%3A%21t%2Cvalue%3A0%29%2Ctime%3A%28from%3Anow-15m%2Cto%3Anow%29%29%26_a%3D%28description%3A%27%27%2Cfilters%3A%21%28%29%29'
```

Comment:
@restrry You are correct query comes before fragment in "regular" URLs, but in angular it's the other way around when using hash routing:
* Docs https://code.angularjs.org/1.8.0/docs/guide/$location#hashbang-and-html5-modes
* Our config https://github.com/elastic/kibana/blob/e976bddf81d5c811daa47f703fb474d3c601bec3/src/plugins/kibana_legacy/public/angular/angular_config.tsx#L137

However I'm not sure whether we still need this - it seems like we switched to transport this information using the history state and this is just a leftover. @ThomThomson @majagrubic can you please verify

Reply:
Thanks for bringing this up. AFAIK, this is not used anymore and should be safe to remove entirely.

Reply:
@majagrubic is it okay if I merge this PR and you remove id in a follow-up? I still need an approval